### PR TITLE
fix: using simple string match replace regex match for contains udf

### LINKED ISSUE #12930: there is something wrong using contains udf
### Describe the bug

Here, I want to find whether a string contain a substring. I using contain udf. I run below case, I got error
```shell
> select contains(lower('$SE=@(('), '$se=@(');
Execution error: Regular expression did not compile: Syntax(
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
regex parse error:
    $se=@(
         ^
error: unclosed group
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
)
>
```
I'm confused about why regex compilation is needed in the contains UDF. The contains UDF should perform a simple string match rather than using regex_match, according to the documentation on [contains](https://datafusion.apache.org/user-guide/sql/scalar_functions_new.html#contains) and its name.

### To Reproduce

_No response_

### Expected behavior

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by zhuliquan**
## Which issue does this PR close?

Closes #12930.

### CODE REVIEW

File: datafusion/functions/src/string/contains.rs [hunk 1]
```diff
@@ -108,34 +108,19 @@ pub fn contains(args: &[ArrayRef]) -> Result<ArrayRef, DataFusionError> {
         (Utf8View, Utf8View) => {
             let mod_str = args[0].as_string_view();
             let match_str = args[1].as_string_view();
-            let res = regexp_is_match::<
-                StringViewArray,
-                StringViewArray,
-                GenericStringArray<i32>,
-            >(mod_str, match_str, None)?;
-
+            let res = arrow_contains(mod_str, match_str)?;
```
Comment (Dandandan) [inline datafusion/functions/src/string/contains.rs:111]: Could you update the doc for `contains`?

Comment (zhuliquan) [reply datafusion/functions/src/string/contains.rs:111]: yeah, I update doc of `contains`
https://github.com/zhuliquan/datafusion/blob/5d0d86a0e0ede41530d2b2b587cdb6f96b7da499/datafusion/functions/src/string/contains.rs#L105

**Comment by Omega359** [comment]
ci failure looks unrelated to changes.

**Comment by zhuliquan** [comment]
> ci failure looks unrelated to changes.
yeah, I add test case for retring ci.😂

**Comment by Omega359** [comment]
LGTM

**Review by alamb — APPROVED** [review]
Thanks @zhuliquan for the contribution and @Omega359 and @findepi for the review

I also double checked that this is covered by @goldmedal 's very nice string_query.slt: https://github.com/apache/datafusion/blob/4534a285c649be8fe7dde56b25078d77d82cf002/datafusion/sqllogictest/test_files/string/string_query.slt.part#L421-L434

**Comment by alamb** [comment]
🚀

**Pull request merged**